packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -16,6 +16,19 @@
 
 import { diag, SpanAttributeValue, SpanAttributes } from '@opentelemetry/api';
 
+function serializeNonPrimitiveAttributeValue(
+  key: string,
+  value: unknown
+): SpanAttributeValue | undefined {
+  if (typeof value === 'object' && Array.isArray(value)) {
```

Comment:
Did you mean to make this an `||`?

Reply:
If this is truly meant to be `&&` then it is not needed to check typeof at all. `Array.isArray` is safe for all inputs